### DISCUSSION TITLE
Performance improvements to trackpad zoom

### DISCUSSION THREAD

**Discussion by ffxsam** [katspaugh/wavesurfer.js#3393] [General]
Per https://github.com/katspaugh/wavesurfer.js/pull/3388#issuecomment-1849083715

Zooming in/out is incredibly CPU-intensive. I'm wondering if we should implement a zoom with detents/"notches" instead.

CC @katspaugh @cmorbitzer

**Comment by katspaugh** [comment]
We could probably add a 300-500-ms debounce.

We can reuse the `createDelay` function used to debounce the window resize handler here: https://github.com/katspaugh/wavesurfer.js/blob/main/src/renderer.ts#L103-L108

**Comment by ffxsam** [comment]
Actually, I'm not sure it's about debouncing or throttling. I tried that, and it made it pretty sluggish. I'm gonna take another approach, will open a PR shortly.

**Reply by katspaugh** [reply]
True, throttling. I always mix up the two.
